lib/logproto/logproto-buffered-server.c

```diff
@@ -42,11 +42,16 @@ typedef struct _BufferedServerBookmarkData
 LogProtoBufferedServerState *
 log_proto_buffered_server_get_state(LogProtoBufferedServer *self)
 {
-  if (self->persist_state)
+  if (self->persist_state && self->persist_handle)
     {
-      g_assert(self->persist_handle != 0);
+      if (G_UNLIKELY(self->persist_handle == 0))
```

Comment:
self->state1 is intended for logproto-buffered cases when there is no need to store state in persistent storage, in memory state is enough: for example tcp buffering.
Instead of this, when opening a file in the wildcard file source case, we should return with error.